src/main/java/core/basesyntax/strategy/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.dao.FruitDaoImpl;
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    private final FruitDao dao = new FruitDaoImpl();
+
+    @Override
+    public void handle(FruitTransaction transaction) {
+        List<FruitTransaction> balance
+                = dao.getByOperation(FruitTransaction.Operation.BALANCE);
+        for (FruitTransaction balanceFruit : balance) {
+            if (balanceFruit.getFruit().equals(transaction.getFruit())) {
+                int balanceFruitQuantity = balanceFruit.getQuantity();
+                int quantityToRemove = transaction.getQuantity();
+                if (balanceFruitQuantity >= quantityToRemove) {
+                    balanceFruit.setQuantity(balanceFruitQuantity - quantityToRemove);
+                }
```

Comment:
you can add your own message for an exception, but it should be with a good description 
```suggestion
                if (balanceFruitQuantity < quantityToRemove) {
                    throw new RuntimeException("Cannot purchase " + quantityToRemove + transaction.getFruit() +  " fruit, cause shop has only " + balanceFruitQuantity);
                }
                balanceFruit.setQuantity(balanceFruitQuantity - quantityToRemove);
```